cli/npm/resolvers/local.rs

```diff
@@ -323,9 +323,39 @@ fn symlink_package_dir(
 
   // need to delete the previous symlink before creating a new one
   let _ignore = fs::remove_dir_all(new_path);
+
+  #[cfg(windows)]
+  return junction_or_symlink_dir(old_path, new_path);
+  #[cfg(not(windows))]
   fs_util::symlink_dir(old_path, new_path)
 }
 
+#[cfg(windows)]
+fn junction_or_symlink_dir(
+  old_path: &Path,
+  new_path: &Path,
+) -> Result<(), AnyError> {
+  // Use junctions because they're supported on ntfs file systems without
+  // needing to elevate privileges on Windows
+  match junction::create(old_path, new_path) {
+    Ok(()) => Ok(()),
+    Err(junction_err) => {
+      match fs_util::symlink_dir(old_path, new_path) {
+        Ok(()) => Ok(()),
+        Err(symlink_err) => bail!(
+          concat!(
+            "Attempted to junction and failed. Then attempted to symlink and also failed.\n\n",
+            "Junction error:\n{:#}\n\nSymlink error:\n{:#}\n\n",
+            "Maybe try enabling developer mode on your machine?"
```

Comment:
Not the most reassuring error message 😄 Can we use more direct wording?

Reply:
I tweaked it a bit. Let me know if you have any suggestions.

Reply:
I was thinking of something along the lines of replacing:

"Maybe try enabling developer mode on your machine?"

with:

"Please try enabling developer mode on your machine. If you continue to experience this error, please open an issue."

Reply:
I'm just going to remove this line because it actually might not be the right solution.